src/main/scala/com/cognite/spark/datasource/TimeSeriesRelation.scala

```diff
@@ -87,13 +91,42 @@ class TimeSeriesRelation(config: RelationConfig)(val sqlContext: SQLContext)
 
   private def postRows(rows: Seq[Row]): IO[Unit] = {
     val timeSeriesItems = rows.map(r => fromRow[PostTimeSeriesItem](r))
-    post(config.apiKey, baseTimeSeriesUrl(config.project), timeSeriesItems, maxRetries)
-      .map(item => {
+    putOr(config.apiKey, baseTimeSeriesUrl(config.project), timeSeriesItems, maxRetries) {
+      case Response(Right(body), StatusCodes.NotFound, _, _, _) =>
+        decode[Error[TimeSeriesNamesNotFound]](body) match {
+          case Right(conflict) => resolveConflict(timeSeriesItems, conflict.error)
+          case Left(error) => IO.raiseError(error)
+        }
+    }.flatTap { _ =>
+      IO {
         if (config.collectMetrics) {
           timeSeriesCreated.inc(rows.length)
         }
-        item
-      })
+      }
+    }
+  }
+
+  def resolveConflict(timeSeriesItems: Seq[PostTimeSeriesItem], timeSeriesConflict: TimeSeriesNamesNotFound): IO[Unit] = {
+    implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+    val notExistingNames = timeSeriesConflict.notFound
```

Comment:
Call it `notFound` instead, it's names today but it will be "external id" in the future it seems.